test/e2e/metadata_update.test.ts

```diff
@@ -1,48 +1,48 @@
 import { expect } from "chai"
-import { randomUUID } from "crypto"
-import { connect, Connection } from "../../src"
+import { Connection, ListenersParams } from "../../src"
 import { MetadataUpdateResponse } from "../../src/responses/metadata_update_response"
+import { createConnection, createPublisher, createStreamName } from "../support/fake_data"
 import { Rabbit } from "../support/rabbit"
 import { eventually } from "../support/util"
 
 describe("update the metadata from the server", () => {
   const rabbit = new Rabbit()
   let connection: Connection
+  let streamName: string
   const metadataUpdateResponses: MetadataUpdateResponse[] = []
 
   beforeEach(async () => {
     metadataUpdateResponses.length = 0
-    connection = await connect({
-      hostname: "localhost",
-      port: 5552,
-      username: "rabbit",
-      password: "rabbit",
-      vhost: "/",
-      frameMax: 0, // not used
-      heartbeat: 0, // not used
-      listeners: { metadata_update: (data) => metadataUpdateResponses.push(data) },
-    })
+    const listeners: ListenersParams = {
+      metadata_update: (data) => metadataUpdateResponses.push(data),
+      credit: (_data) => console.error("Subscribe server error"),
```

Comment:
Are you sure we want to add this listener here? why don't simply use the

Reply:
I don’t think it’s necessary give the ability to add this type of listener on the creation. I would keep this listener more private as possible. WDYT?